docs/index.md

```diff
@@ -0,0 +1,123 @@
+
+## Installation
+
+Apply keeper _after_ the `android {}` block in your application's build.gradle file and configure
+via the `keeper {}` extension.
+
+```groovy
+buildscript {
+  repositories {
+    mavenCentral()
+    // ...
+  }
+
+  dependencies {
+    // ...
+    classpath "com.slack.keeper:keeper:x.y.z"
+  }
+
+}
+
+android {
+  // ...
+}
+
+apply plugin: "com.slack.keeper"
+keeper {
+  /**
+   * The name of the androidTest variant to infer for. This should include the "AndroidTest"
+   * suffix. Required.
+   */
+  androidTestVariant = "yourAndroidTestVariantHere" // Required
+
+  /** The name of the application variant to infer for. Required. */
+  appVariant = "yourReleaseVariantHere"
+
+  /** R8 version. Must be a tag. Optional, default defined below. */
+  r8Version = "1.6.53"
+
+  /**
+   * Optional custom jvm arguments to pass into the R8 `PrintUses` execution. Useful if you want
+   * to enable debugging in R8. Optional, default is empty.
+   *
+   * Example: `listOf("-Xdebug", "-Xrunjdwp:transport=dt_socket,address=5005,server=y,suspend=y")`
+   */
+  r8JvmArgs = []
+}
+```
+
+### Dynamic Configuration
+
+If you want to dynamically configure this, one approach we use internally is to define a `buildVariant`
+gradle property and use its value to gate application of this plugin and which variant to use.
+
+```groovy
+if (project.hasProperty("buildVariant")) {
+  String buildVariant = project.getProperty("buildVariant")
+  apply plugin: 'com.slack.keeper
+
+  keeper {
+    androidTestVariant = "${buildVariant}AndroidTest"
+    appVariant = buildVariant
+  }
+}
+```
+
+Snapshots of the development version are available in [Sonatype's `snapshots` repository][snapshots].
+
+## Under the hood
+
+The general logic flow:
+- Create a custom `r8` configuration for the R8 dependency.
+- Register three jar tasks. One for all the classes in the app variant, one for all the classes in
+  the androidTest variant, and one copy of the compiled android.jar for classpath linking. This
+  will use their variant-provided `JavaCompile` tasks and `KotlinCompile` tasks if available.
+- Register a [`inferAndroidTestUsage`](https://github.com/slackhq/keeper/blob/master/keeper-gradle-plugin/src/main/kotlin/com/slack/keeper/InferAndroidTestKeepRules.kt)
+  task that plugs the two aforementioned jars into R8's `PrintUses` CLI and outputs the inferred
+  proguard rules into a new intermediate `.pro` file.
+- Finally - the generated file is wired in to Proguard/R8 via private `ProguardConfigurable` API.
+  This works by looking for the relevant `TransformTask` that uses it.
+
+Appropriate task dependencies (via inputs/outputs, not `dependsOn`) are set up, so this is
+automatically run as part of the target app variant's full minified APK.
+
+The tasks themselves take roughly ~20 seconds total extra work in our Slack android app, with the
+infer and app jar tasks each taking around 8-10 seconds and the androidTest jar taking around 2 seconds.
```

Comment:
Do you want to document the side-effect that this will run on external release variants? Or you think that you'll be able to solve that issue prior to release?

Reply:
Eventually yes, but the public repo doesn’t have that change upstreamed to it yet